src/app/dataScripts/httpsRequestToCallback.ts

```diff
@@ -0,0 +1,47 @@
+import https from 'node:https';
+
+export default function httpsRequestToCallback({
+    url,
+    options = {},
+    body,
+    onDataResponse,
+}: {
+    url: string;
+    options?: https.RequestOptions;
+    body?: string;
+    onDataResponse: any;
```

Comment:
Best to avoid using `any` if possible, though in this case the type of the json is unknown. Still, you can at least type this parameter as a function:
```suggestion
    onDataResponse: (parsedResult: any) => void
```
Even better would be to add a generic type to this function so external users could set the type they expect to be returned.

Reply:
Nice, I went with your suggestions here and added as part of 4825319db98d00501351e2a72d31bd97a8ee5a1e